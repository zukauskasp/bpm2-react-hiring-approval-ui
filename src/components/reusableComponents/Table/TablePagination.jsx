import React from "react";
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Icon from "../Icon/Icon";

const Pagination = ({ page, resultsCount, rowsPerPage, rowsPerPageOptions, filteredDataCount, setPage, setRowsPerPage, onChangeRowsPerPage }) => {

    const classes = useStyles();

    return (
        <Grid container direction="row" justify="space-between" alignItems="center" className={`${classes.actionsWrapper}`} >
            <Grid item >
                <span className={`${classes.paginationSpan}`}>{filteredDataCount} out of {resultsCount}</span>
            </Grid>
            <Grid item>
                {/*Example output: <...2 3 4...> */}
                {/* Display arrow left if there are at least one page to the left*/}
                {page - 1 > 0 ? <Icon alt="" className={classes.paginationIcon} icon='back' onClick={() => setPage(page - 1)} /> : null}
                {/* Display multiple dots on the left if there are at least two pages to the left*/}
                {page - 2 > 0 ? <span className={`${classes.paginationSpan}`}>...</span> : null}
                {/* Display page number to the left if there is at least one page to the left*/}
                {page - 1 > 0 ? <span className={`${classes.paginationSpan}`} onClick={() => setPage(page - 1)}>{page - 1}</span> : null}
                {/* Display current page*/}
                <span className={`${classes.paginationSpan} ${classes.paginationSpanSelected}`}>{page}</span>
                {/* Display page number to the right if there is at least one page to the right*/}
                {resultsCount / rowsPerPage > page ? <span className={`${classes.paginationSpan}`} onClick={() => setPage(page + 1)}>{page + 1}</span> : null}
                {/*  Display multiple dots on the right if there are at least two pages to the right*/}
                {resultsCount / rowsPerPage > page + 1 ? <span className={`${classes.paginationSpan}`}>...</span> : null}
                {/*  Display arrow right if there are at least one page to the right*/}
                {resultsCount / rowsPerPage > page ? <Icon alt="" className={classes.paginationIcon} icon='next' onClick={() => setPage(page + 1)} /> : null}
            </Grid>
            <Grid item >
                {rowsPerPageOptions.map((value, index) => (
                    <span
                        key={index}
                        className={`${classes.paginationSpan} ${rowsPerPage === value ? classes.paginationSpanSelected : null}`}
                        onClick={() => { setRowsPerPage(value); onChangeRowsPerPage(value); setPage(1) }}
                    >{value}
                    </span>
                ))}
            </Grid>
        </Grid>
    )
}

Pagination.propTypes = {
    page: PropTypes.number.isRequired,
    resultsCount: PropTypes.number.isRequired,
    rowsPerPage: PropTypes.number.isRequired,
    rowsPerPageOptions: PropTypes.arrayOf(PropTypes.number),
    filteredDataCount: PropTypes.number.isRequired,
    setPage: PropTypes.func.isRequired,
    setRowsPerPage: PropTypes.func.isRequired,
    onChangeRowsPerPage: PropTypes.func.isRequired
}

Pagination.defaultProps = {
    page: 1,
    rowsPerPage: 5,
    rowsPerPageOptions: [5, 10, 20],
}

export default Pagination;

const useStyles = makeStyles({
    actionsWrapper: {
        padding: "15px 10px",
        boxSizing: "border-box"
    },
    paginationSpan: {
        fontFamily: 'Danske Text v2',
        fontWeight: '600',
        fontSize: "15px",
        padding: '0 5px',
        color: "#93989C",
        cursor: "pointer",

    },
    paginationSpanSelected: {
        color: "#003755",
    },
    paginationIcon: {
        cursor: "pointer",
        height: '1.2em',
        margin: '5px 0 0 0',
        verticalAlign: 'text-bottom'
    }
});



