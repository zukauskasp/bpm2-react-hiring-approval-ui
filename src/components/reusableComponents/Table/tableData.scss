.full-width-table-cell {
    width: 99%;
}