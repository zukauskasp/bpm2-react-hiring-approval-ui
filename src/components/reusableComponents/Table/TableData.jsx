import React from "react";
import './tableData.scss';
import { withStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';

export default function TableData(props) {
    return (
        <StyledTableCell {...props}>{props.children}</StyledTableCell>
    )
}
const StyledTableCell = withStyles(() => ({
    root: {
        fontFamily: "Danske Text v2",
        paddingRight: "40px",
        fontSize: '14px',
        color: "#003755",
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
        overflow: 'hidden',
        maxWidth: "200px",

    },
    head: {
        fontWeight: "600",
    },
    body: {
        fontWeight: "400",
        paddingRight: "40px",

        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis',
        overflow: 'hidden',
    },
    sizeSmall: {
        width: '55px',
        paddingRight: '22px',
        '&:last-child': {
            paddingRight: '24px'
        },
        '&:first-child': {
            paddingRight: '0px'
        }
    }

}))(TableCell);
