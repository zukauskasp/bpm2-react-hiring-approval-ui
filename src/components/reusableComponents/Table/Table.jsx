import React, { useState } from "react";
import { makeStyles, withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import Checkbox from "../Checkbox/Checkbox";
import TableData from "./TableData";
import PaginationActions from "./TablePagination";
import Icon from "../Icon/Icon";

import Paper from '@material-ui/core/Paper';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableContainer from '@material-ui/core/TableContainer';
import TableRow from '@material-ui/core/TableRow';
import TableHead from '@material-ui/core/TableHead';


const CustomTable = ({ selectionTable, expandableTable, data, setData, titles, copyAction, editAction, deleteAction }) => {

    const classes = useStyles();
    const [page, setPage] = useState(1);
    const [rowsPerPage, setRowsPerPage] = useState(5);
    const [openExpandableRowId, setOpenExpandableRowId] = useState(null);

    const handleChangeRowsPerPage = (value) => {
        setRowsPerPage(value);
        setPage(1);
    };

    const setChecked = (id) => {
        let newData = [...data];
        newData.map(row => row.id === id ? row.isChecked = !row.isChecked : null)
        setData(newData)
    }

    const toggleTableRow = id => id === openExpandableRowId ? setOpenExpandableRowId(null) : setOpenExpandableRowId(id);

    return (
        <Paper elevation={0} >
            <TableContainer >
                <Table className={classes.table} >
                    {titles && titles.length > 0 ?
                        <TableHead>
                            <TableRow>
                                {selectionTable ? <TableData size="small" padding="checkbox" /> : null}
                                {expandableTable ? <TableData size="small" padding="checkbox" /> : null}
                                {titles.map((value, index) => (
                                    <TableData key={index} title={value} >{value}</TableData>
                                ))}
                                {copyAction ? <TableData /> : null}
                                {editAction ? <TableData /> : null}
                                {deleteAction ? <TableData /> : null}
                            </TableRow>
                        </TableHead>
                        : null}

                    <TableBody>
                        {data.slice((page - 1) * rowsPerPage, (page - 1) * rowsPerPage + rowsPerPage).map((row, key) => (
                            <React.Fragment key={key}>

                                {/*Table row start */}
                                <StyledRow key={row.id}>
                                    {selectionTable ? // show checkboxes if true
                                        <TableData size="small" ><Checkbox value={row.isChecked} checked={row.isChecked} onChange={() => { setChecked(row.id) }} /></TableData>
                                        : null}
                                    {expandableTable ? // show expand icon if true
                                        <TableData size="small" padding="checkbox" >
                                            {expandableTable && row.expandableData !== "" ?
                                                <Icon
                                                    icon={row.id !== openExpandableRowId ? 'expand' : 'collapse'}
                                                    className={classes.icon}
                                                    onClick={() => toggleTableRow(row.id)}
                                                />
                                                : null}
                                        </TableData>
                                        : null}

                                    {/*Display data start*/}
                                    {Object.keys(row.data).map((value, index) => {
                                        return <TableData key={index}
                                            className={index + 1 === Object.keys(row.data).length ? "full-width-table-cell" : ""}
                                            title={row.data[value]}>
                                            {row.data[value]}
                                        </TableData>
                                    })}
                                    {/*Display data end*/}

                                    {copyAction ? <TableData size="small"><Icon alt="" className={classes.icon} icon='duplicate' onClick={() => copyAction(row)} /> </TableData> : null}
                                    {editAction ? <TableData size="small"><Icon alt="" className={classes.icon} icon='edit' onClick={() => editAction(row, row.id)} /></TableData> : null}
                                    {deleteAction ? <TableData size="small"><Icon alt="" className={classes.icon} icon='trash' onClick={() => deleteAction(row.id)} /></TableData> : null}
                                </StyledRow>
                                {/*Table row end */}

                                {/*Expandable row start */}
                                {expandableTable && row.id === openExpandableRowId && row.expandableData !== "" ?
                                    <StyledRow key={`${row.id}_expanded`}>
                                        <TableData colSpan="100%" style={{ textOverflow: 'initial', whiteSpace: 'normal' }} >{row.expandableData}</TableData>
                                    </StyledRow>
                                    : null}
                                {/*Expandable row end*/}
                            </React.Fragment>
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>
            <PaginationActions
                resultsCount={data.length}
                rowsPerPage={rowsPerPage}
                setRowsPerPage={setRowsPerPage}
                page={page}
                setPage={setPage}
                onChangeRowsPerPage={handleChangeRowsPerPage}
                filteredDataCount={data.slice((page - 1) * rowsPerPage, (page - 1) * rowsPerPage + rowsPerPage).length}
            />
        </Paper>
    );
}

CustomTable.propTypes = {
    selectionTable: PropTypes.bool,
    expandableTable: PropTypes.bool,
    data: PropTypes.arrayOf(PropTypes.exact({
        id: PropTypes.number,
        isChecked: PropTypes.bool,
        data: PropTypes.object,
        expandableData: PropTypes.string
    })),
    setData: function (props, propName, componentName) {
        if ((props['selectionTable'] === true && (props[propName] === undefined || typeof (props[propName]) !== 'function'))) {
            return new Error('setData function is required for selectionTable');
        }
    },
    titles: PropTypes.array.isRequired,
    copyAction: PropTypes.func,
    editAction: PropTypes.func,
    deleteAction: PropTypes.func,
}

CustomTable.defaultProps = {
    data: []
}

export default CustomTable;

const useStyles = makeStyles({
    table: {
        minWidth: 650,
    },
    icon: {
        cursor: 'pointer'
    }
});

const StyledRow = withStyles(() => ({
    root: {
        transitionDuration: ".2s",
        "&:hover": {
            background: "#F0F7FA",
        },
    },
}))(TableRow);

