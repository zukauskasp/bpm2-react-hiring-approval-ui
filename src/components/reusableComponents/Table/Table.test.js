import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';

import Table from './Table';
import Pagination from "./TablePagination";

describe('Table', () => {

    let wrapper;
    let mount;
    let shallow;
    let list = [
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "SENIOR", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "JUNIOR", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "MID", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer khlgk jgkfld", departament: "Test departament", position: "MID", salaryRange: "2000-3000 EUR 111111111", date: "2020-03-20" },
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "SENIOR", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "JUNIOR", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer", departament: "Test departament", position: "MID", salaryRange: "2000-3000 EUR", date: "2020-03-20" },
        { jobTitle: "Softwere engineer khlgk jgkfld", departament: "Test departament", position: "MID", salaryRange: "2000-3000 EUR 111111111", date: "2020-03-20" },
    ];
    let data = [];
    const titles = ["Job description", "Departament longer title", "Position", "Salary (range)", "Start date"];

    beforeAll(() => {
        shallow = createShallow();
        mount = createMount();
        list.map((value, key) => data.push({ id: key, isChecked: false, data: value, expandableData: "" }));

    });

    const props = {
        selectionTable: true,
        expandableTable: false,
        data: data,
        setData: () => { },
        titles: titles,
        copyAction: null,
        editAction: () => { },
        deleteAction: () => { }
    }

    const paginationProps = {
        resultsCount: list.length,
        filteredDataCount: 5,
        onChangeRowsPerPage: () => { },
        setRowsPerPage: () => { },
        setPage: () => { }
    }

    beforeEach(() => {
        wrapper = shallow(<Table {...props} />);
    })

    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        expect(shallow(<Table {...props} />).debug()).toMatchSnapshot();
    });

    it('expect to render table', (() => {
        expect(wrapper.length).toEqual(1);
    }));

    it('expect to render correct props', () => {
        expect(mount(<Table {...props} />).props()).toEqual(props);
    });

    it('expect to render default props', () => {
        const mounted = mount(<Table titles={props.titles} setData={props.setData} />);

        expect(mounted.props().data).toEqual([]);
    });

    it('expect to render list', () => {
        const mounted = mount(<Table {...props} />);

        expect(mounted.prop('data').length).toEqual(list.length);
    });


    it('expect to render pagination', () => {
        const wrapper = shallow(<Pagination {...paginationProps} />);

        expect(wrapper.length).toEqual(1);
    })


    it('expect to render default props', () => {
        const mounted = mount(<Pagination {...paginationProps} />);

        expect(mounted.props().page).toEqual(1);
        expect(mounted.props().rowsPerPage).toEqual(5);
        expect(mounted.props().rowsPerPageOptions).toEqual([5, 10, 20]);
    });


    it('expect to render correct rows count according to pagination rules', () => {
        var length = list.length <= paginationProps.filteredDataCount ? list.length : paginationProps.filteredDataCount;

        expect(wrapper.childAt(0).childAt(0).childAt(1).children().length).toEqual(length);
    });


    it('expect to render icons correncly', () => {
        const mounted = mount(<Pagination {...paginationProps} />);
        const wrapper = shallow(<Pagination {...paginationProps} />);
        const page = mounted.prop('page');
        const resultsCount = mounted.prop('resultsCount');
        const rowsPerPage = mounted.prop('rowsPerPage')

        if (page - 1 > 0) {
            expect(wrapper.find({ 'icon': 'back' }).length).toEqual(1);
        }

        if (resultsCount / rowsPerPage > page) {
            expect(wrapper.find({ 'icon': 'next' }).length).toEqual(1);
        }

    });

})