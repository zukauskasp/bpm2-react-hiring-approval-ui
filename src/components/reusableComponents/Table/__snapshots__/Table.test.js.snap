// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table expect to match latest snapshot 1`] = `
"<WithStyles(ForwardRef(Paper)) elevation={0}>
  <WithStyles(ForwardRef(TableContainer))>
    <WithStyles(ForwardRef(Table)) className=\\"makeStyles-table-1\\">
      <WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableRow))>
          <TableData size=\\"small\\" padding=\\"checkbox\\" />
          <TableData title=\\"Job description\\">
            Job description
          </TableData>
          <TableData title=\\"Departament longer title\\">
            Departament longer title
          </TableData>
          <TableData title=\\"Position\\">
            Position
          </TableData>
          <TableData title=\\"Salary (range)\\">
            Salary (range)
          </TableData>
          <TableData title=\\"Start date\\">
            Start date
          </TableData>
          <TableData />
          <TableData />
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableBody))>
        <WithStyles(WithStyles(ForwardRef(TableRow)))>
          <TableData size=\\"small\\">
            <CustomCheckbox value={false} checked={false} onChange={[Function: onChange]} readonly={false} errorMessage=\\"\\" label=\\"\\" />
          </TableData>
          <TableData className=\\"\\" title=\\"Softwere engineer\\">
            Softwere engineer
          </TableData>
          <TableData className=\\"\\" title=\\"Test departament\\">
            Test departament
          </TableData>
          <TableData className=\\"\\" title=\\"SENIOR\\">
            SENIOR
          </TableData>
          <TableData className=\\"\\" title=\\"2000-3000 EUR\\">
            2000-3000 EUR
          </TableData>
          <TableData className=\\"full-width-table-cell\\" title=\\"2020-03-20\\">
            2020-03-20
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"edit\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"trash\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
        </WithStyles(WithStyles(ForwardRef(TableRow)))>
        <WithStyles(WithStyles(ForwardRef(TableRow)))>
          <TableData size=\\"small\\">
            <CustomCheckbox value={false} checked={false} onChange={[Function: onChange]} readonly={false} errorMessage=\\"\\" label=\\"\\" />
          </TableData>
          <TableData className=\\"\\" title=\\"Softwere engineer\\">
            Softwere engineer
          </TableData>
          <TableData className=\\"\\" title=\\"Test departament\\">
            Test departament
          </TableData>
          <TableData className=\\"\\" title=\\"JUNIOR\\">
            JUNIOR
          </TableData>
          <TableData className=\\"\\" title=\\"2000-3000 EUR\\">
            2000-3000 EUR
          </TableData>
          <TableData className=\\"full-width-table-cell\\" title=\\"2020-03-20\\">
            2020-03-20
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"edit\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"trash\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
        </WithStyles(WithStyles(ForwardRef(TableRow)))>
        <WithStyles(WithStyles(ForwardRef(TableRow)))>
          <TableData size=\\"small\\">
            <CustomCheckbox value={false} checked={false} onChange={[Function: onChange]} readonly={false} errorMessage=\\"\\" label=\\"\\" />
          </TableData>
          <TableData className=\\"\\" title=\\"Softwere engineer\\">
            Softwere engineer
          </TableData>
          <TableData className=\\"\\" title=\\"Test departament\\">
            Test departament
          </TableData>
          <TableData className=\\"\\" title=\\"MID\\">
            MID
          </TableData>
          <TableData className=\\"\\" title=\\"2000-3000 EUR\\">
            2000-3000 EUR
          </TableData>
          <TableData className=\\"full-width-table-cell\\" title=\\"2020-03-20\\">
            2020-03-20
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"edit\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"trash\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
        </WithStyles(WithStyles(ForwardRef(TableRow)))>
        <WithStyles(WithStyles(ForwardRef(TableRow)))>
          <TableData size=\\"small\\">
            <CustomCheckbox value={false} checked={false} onChange={[Function: onChange]} readonly={false} errorMessage=\\"\\" label=\\"\\" />
          </TableData>
          <TableData className=\\"\\" title=\\"Softwere engineer khlgk jgkfld\\">
            Softwere engineer khlgk jgkfld
          </TableData>
          <TableData className=\\"\\" title=\\"Test departament\\">
            Test departament
          </TableData>
          <TableData className=\\"\\" title=\\"MID\\">
            MID
          </TableData>
          <TableData className=\\"\\" title=\\"2000-3000 EUR 111111111\\">
            2000-3000 EUR 111111111
          </TableData>
          <TableData className=\\"full-width-table-cell\\" title=\\"2020-03-20\\">
            2020-03-20
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"edit\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"trash\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
        </WithStyles(WithStyles(ForwardRef(TableRow)))>
        <WithStyles(WithStyles(ForwardRef(TableRow)))>
          <TableData size=\\"small\\">
            <CustomCheckbox value={false} checked={false} onChange={[Function: onChange]} readonly={false} errorMessage=\\"\\" label=\\"\\" />
          </TableData>
          <TableData className=\\"\\" title=\\"Softwere engineer\\">
            Softwere engineer
          </TableData>
          <TableData className=\\"\\" title=\\"Test departament\\">
            Test departament
          </TableData>
          <TableData className=\\"\\" title=\\"SENIOR\\">
            SENIOR
          </TableData>
          <TableData className=\\"\\" title=\\"2000-3000 EUR\\">
            2000-3000 EUR
          </TableData>
          <TableData className=\\"full-width-table-cell\\" title=\\"2020-03-20\\">
            2020-03-20
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"edit\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
          <TableData size=\\"small\\">
            <Icon alt=\\"\\" className=\\"makeStyles-icon-2\\" icon=\\"trash\\" onClick={[Function: onClick]} color=\\"dark\\" />
          </TableData>
        </WithStyles(WithStyles(ForwardRef(TableRow)))>
      </WithStyles(ForwardRef(TableBody))>
    </WithStyles(ForwardRef(Table))>
  </WithStyles(ForwardRef(TableContainer))>
  <Pagination resultsCount={8} rowsPerPage={5} setRowsPerPage={[Function: bound _dispatchAction]} page={1} setPage={[Function: bound _dispatchAction]} onChangeRowsPerPage={[Function: handleChangeRowsPerPage]} filteredDataCount={5} rowsPerPageOptions={{...}} />
</WithStyles(ForwardRef(Paper))>"
`;
